### PR TITLE
Expose `idx()` method on `Guard`

### PR DESCRIPTION
**Pull request by davidbarsky**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -206,6 +206,16 @@ pub struct Guard<'a, T, C: cfg::Config = DefaultConfig> {
     key: usize,
 }
 
+impl<'a, T, C> Guard<'a, T, C>
+where
+    C: cfg::Config,
+{
+    /// Returns the idx of the guard.
```
Comment (hawkw) [inline src/lib.rs]:
```suggestion
    /// Returns the index of the guard.
```

Comment (hawkw) [reply src/lib.rs]: Also, it's not really the index of the _guard_ so much as the index of the _slot_ that the guard represents an access to...maybe this should be reworded.

File: src/lib.rs [hunk 2]
```diff
@@ -206,6 +206,16 @@ pub struct Guard<'a, T, C: cfg::Config = DefaultConfig> {
     key: usize,
 }
 
+impl<'a, T, C> Guard<'a, T, C>
```
Comment (hawkw) [inline src/lib.rs]: Can we please move this impl block [down here](https://github.com/hawkw/sharded-slab/blob/2691621d04d0dbe24fd790d94aaed1f4f1eb8844/src/lib.rs#L600-L642) with the rest of the impls on `Guard`?

File: src/lib.rs [hunk 3]
```diff
@@ -206,6 +206,16 @@ pub struct Guard<'a, T, C: cfg::Config = DefaultConfig> {
...
+where
+    C: cfg::Config,
+{
+    /// Returns the idx of the guard.
+    pub fn idx(&self) -> usize {
```
Comment (hawkw) [inline src/lib.rs]: I'm not sure if I like this naming...elsewhere, the API tends to refer to slab indices as "keys". The `slab` crate has [a similar method](https://github.com/carllerche/slab/blob/f1327f72fb14fb387e07a6123300eb7a9da7c4e8/src/lib.rs#L1188-L1211), which it calls `key`...

**Comment by davidbarsky** [comment]
Thanks for the feedback. I'm closing this PR and opening a new one so avoid rebasing.

**Pull request closed without merging**